lib/middleman-prismic/preview.rb

```diff
@@ -0,0 +1,22 @@
+module Middleman
+  module Prismic
+    class Preview
+      def initialize(app, options={})
+        @app = app
+      end
+
+      def call(env)
+        req = ::Rack::Request.new(env)
+        if req.path =~ /^\/preview/
+          token = req.params["token"]
+
+          Kernel.system "middleman prismic --ref #{token}"
```

Comment:
This is interpolating unescaped user input into a system command. You'll want to verify that the token follows a known pattern (like alphanumeric) or escape it using a tool like [Terrapin](https://github.com/thoughtbot/terrapin).

Reply:
[`ShellWords.escape`][1] might be sufficient here.

[1]: http://ruby-doc.org/stdlib-2.0.0/libdoc/shellwords/rdoc/Shellwords.html#method-c-escape

Reply:
Good thinking, joe and george. i hadn't considered that. e94a68c

Reply:
some more cleanup 621d741